[VS Code/Rider][mac/Windows] Hot Reload fails with compilation error when editing a Skia/Native + .NET MAUI Embedding project

### Current behavior 🐛

- When a new project is created using the Recommended template with Skia/Native + MAUI Embedding, Hot Reload (HR) does not function as expected. After making changes to a file and saving it, the HR indicator displays a compilation error, and the app does not reflect updates.
- Enetering and adding element as well throws compilation error
Affected TFMs: Desktop and WebAssembly
Works on TFMs: Android and iOS
sdk/template: 6.0.146
VS code VSIX: 0.20.7

[HotReload.txt](https://github.com/user-attachments/files/20805090/HotReload.txt)
[UnoPlatform- debugger.txt](https://github.com/user-attachments/files/20805088/UnoPlatform-.debugger.txt)
[UnoPlatformLogs.txt](https://github.com/user-attachments/files/20805089/UnoPlatformLogs.txt)


https://github.com/user-attachments/assets/c412b8c2-1fa1-4f1e-a44a-f3e95b05c7e7

### Expected behavior 🎯

1. Create a Skia + Maui Project
2. Build solution for Desktop/ webAssembly
3. Build solution without debugger
4. Make any small changes in 'MainPage.xaml' file (eg: add 123 in PlaceholderText)
5. save changed
6. Observe HR indicator will show 'Compile error'.

### How to reproduce it (as minimally and precisely as possible) 🔬

_No response_

### Workaround 🛠️

_No response_

### Renderer 🎨

- [x] Skia
- [x] Native

### Affected platforms 📱💻🖥️

WebAssembly, Desktop (macOS)

### Uno.Sdk version (and other relevant versions) 📦

Sdk/template: 6.0.146
HD version: 1.13.6

### IDE version 🧑‍💻

1.100.3

### Anything else we need to know? 💬

_No response_

IT does not seem related to VS code, it fails on VS as well :
```
Uno.UI.RemoteControl.HotReload.ClientHotReloadProcessor: Debug: [01-MainPage.xaml] Updating file C:/Users/KunalShah/source/repos/UnoApp7/UnoApp7/Presentation/MainPage.xaml (from:  | to: <Page x:Class="UnoApp7.Presentation.MainPage"
      xmlns="http://schemas.microsoft.com/winfx/2006/.
Uno.UI.RemoteControl.HotReload.ClientHotReloadProcessor: Trace: [01-MainPage.xaml] Successfully updated file on server (Success), waiting for server HR id 1.
Uno.UI.RemoteControl.HotReload.ClientHotReloadProcessor: Debug: [01-MainPage.xaml] Server failed to applied changes in code: Failed.
```
```
​15:50:03.04 [Error] C:\Users\KunalShah\source\repos\UnoApp7\UnoApp7\App.xaml.cs (line 1): error CS0234: The type or namespace name 'Resizetizer' does not exist in the namespace 'Uno' (are you missing an assembly reference?)
15:50:03.04 Invalid changes were found. Please refer to the Error List window to fix those issues.
15:50:03.05 [Error] Project C:\Users\KunalShah\source\repos\UnoApp7\UnoApp7\UnoApp7.csproj (net9.0-desktop {1298D9B8-DFB2-4AD0-88C2-8CBDF49EA89C}) update error: C:\Users\KunalShah\source\repos\UnoApp7\UnoApp7\App.xaml.cs(1,11): error CS0234: The type or namespace name 'Resizetizer' does not exist in the namespace 'Uno' (are you missing an assembly reference?)
15:50:12.52 UnoApp7 (2):18932: The process has exited.
15:50:12.52 UnoApp7 (2): Hot Reload session complete.
```

--- a/src/Uno.UI.RemoteControl.Server.Processors/HotReload/MetadataUpdates/CompilationWorkspaceProvider.cs
+++ b/src/Uno.UI.RemoteControl.Server.Processors/HotReload/MetadataUpdates/CompilationWorkspaceProvider.cs
@@ -34,6 +34,7 @@ namespace Uno.UI.RemoteControl.Host.HotReload.MetadataUpdates
 
 				// Unconditionally enable binlog generation in msbuild. See https://github.com/dotnet/project-system/blob/4210ce79cfd35154dbd858f056bfb9101f290e69/docs/design-time-builds.md?L61
 				Environment.SetEnvironmentVariable("MSBUILDDEBUGENGINE", "1");
+				Environment.SetEnvironmentVariable("MSBuildDebugEngine", "1"); // For case-insensitive environments like macOS
 				Environment.SetEnvironmentVariable("MSBUILDDEBUGPATH", logPath);
 			}
 

--- a/src/Uno.UI.RemoteControl.Server.Processors/HotReload/MetadataUpdates/CompilationWorkspaceProvider.cs
+++ b/src/Uno.UI.RemoteControl.Server.Processors/HotReload/MetadataUpdates/CompilationWorkspaceProvider.cs
@@ -34,7 +34,7 @@ namespace Uno.UI.RemoteControl.Host.HotReload.MetadataUpdates
 
 				// Unconditionally enable binlog generation in msbuild. See https://github.com/dotnet/project-system/blob/4210ce79cfd35154dbd858f056bfb9101f290e69/docs/design-time-builds.md?L61
 				Environment.SetEnvironmentVariable("MSBUILDDEBUGENGINE", "1");
-				Environment.SetEnvironmentVariable("MSBuildDebugEngine", "1"); // For case-insensitive environments like macOS
+				Environment.SetEnvironmentVariable("MSBuildDebugEngine", "1"); // For case-sensitive environments like macOS
 				Environment.SetEnvironmentVariable("MSBUILDDEBUGPATH", logPath);
 			}
 

--- a/src/Uno.UI.RemoteControl.Server.Processors/HotReload/ServerHotReloadProcessor.MetadataUpdate.cs
+++ b/src/Uno.UI.RemoteControl.Server.Processors/HotReload/ServerHotReloadProcessor.MetadataUpdate.cs
@@ -110,6 +110,15 @@ namespace Uno.UI.RemoteControl.Host.HotReload
 						}
 					}
 
+					// Pass the TargetFramework as a temporary property so that we do not force the tfm for all projects, but only the head project
+					// (that references the Dev Server assembly which includes the target file to promotes back the UnoHotReloadTargetFramework as TargetFramework).
+					// This is required to make sure that an application that referencing a class-lib project targeting a different TFM (e.g. net10 while head is net10-desktop)
+					// can still be hot-reloaded.
+					if (properties.Remove("TargetFramework", out var targetFramework))
+					{
+						properties["UnoHotReloadTargetFramework"] = targetFramework;
+					}
+
 					var result = await CompilationWorkspaceProvider.CreateWorkspaceAsync(
 						configureServer.ProjectPath,
 						_reporter,

--- a/src/Uno.UI.RemoteControl.Server.Processors/HotReload/ServerHotReloadProcessor.MetadataUpdate.cs
+++ b/src/Uno.UI.RemoteControl.Server.Processors/HotReload/ServerHotReloadProcessor.MetadataUpdate.cs
@@ -111,8 +111,8 @@ namespace Uno.UI.RemoteControl.Host.HotReload
 					}
 
 					// Pass the TargetFramework as a temporary property so that we do not force the tfm for all projects, but only the head project
-					// (that references the Dev Server assembly which includes the target file to promotes back the UnoHotReloadTargetFramework as TargetFramework).
-					// This is required to make sure that an application that referencing a class-lib project targeting a different TFM (e.g. net10 while head is net10-desktop)
+					// (that references the Dev Server assembly which includes the target file to promote back the UnoHotReloadTargetFramework as TargetFramework).
+					// This is required to make sure that an application referencing a class-lib project targeting a different TFM (e.g. net10 while head is net10-desktop)
 					// can still be hot-reloaded.
 					if (properties.Remove("TargetFramework", out var targetFramework))
 					{

--- a/src/Uno.UI.RemoteControl/buildTransitive/Uno.WinUI.DevServer.props
+++ b/src/Uno.UI.RemoteControl/buildTransitive/Uno.WinUI.DevServer.props
@@ -3,6 +3,7 @@
 
 	<PropertyGroup Condition="'$(UnoIsHotReloadHost)'=='true'">
 		<RuntimeIdentifier Condition="'$(UnoHotReloadRuntimeIdentifier)' != ''">$(UnoHotReloadRuntimeIdentifier)</RuntimeIdentifier>
+		<TargetFramework Condition="'$(UnoHotReloadTargetFramework)' != ''">$(UnoHotReloadTargetFramework)</TargetFramework>
 	</PropertyGroup>
 
 </Project>

